scripts/20newsgroups-replication.py

```diff
@@ -19,43 +21,44 @@ def get_info(path):
     return (docs, targets)
 
 
+class VectorizerType(Enum):
+    TFIDF = 'tfidf'
+    BM25 = 'bm25'
+
+
 if __name__ == '__main__':
-    target_names = [
-        'alt.atheism',
-        'comp.graphics',
-        'comp.os.ms-windows.misc',
-        'comp.sys.ibm.pc.hardware',
-        'comp.sys.mac.hardware',
-        'comp.windows.x',
-        'misc.forsale',
-        'rec.autos',
-        'rec.motorcycles',
-        'rec.sport.baseball',
-        'rec.sport.hockey',
-        'sci.crypt',
-        'sci.electronics',
-        'sci.med',
-        'sci.space',
-        'soc.religion.christian',
-        'talk.politics.guns',
-        'talk.politics.mideast',
-        'talk.politics.misc',
-        'talk.religion.misc',
-    ]
+    parser = argparse.ArgumentParser(
+        description='Replication script of pyserini vectorizer')
+    parser.add_argument('-vectorizer', type=VectorizerType, required=True, help='which vectorizer to use')
+    args = parser.parse_args()
+
+    target_names = ['alt.atheism', 'comp.graphics', 'comp.os.ms-windows.misc', 'comp.sys.ibm.pc.hardware', 'comp.sys.mac.hardware', 'comp.windows.x', 'misc.forsale', 'rec.autos', 'rec.motorcycles', 'rec.sport.baseball',
+                    'rec.sport.hockey', 'sci.crypt', 'sci.electronics', 'sci.med', 'sci.space', 'soc.religion.christian', 'talk.politics.guns', 'talk.politics.mideast', 'talk.politics.misc', 'talk.religion.misc', ]
 
     target_to_index = {t: i for i, t in enumerate(target_names)}
 
     train_docs, train_labels = get_info('./20newsgroups/20news-bydate-train/')
     test_docs, test_labels = get_info('./20newsgroups/20news-bydate-test/')
 
-    vectorizer = TfidfVectorizer(
-        './20newsgroups/lucene-index.20newsgroup.pos+docvectors+raw', min_df=5)
+    lucene_index_path = '20newsgroups/lucene-index.20newsgroup.pos+docvectors+raw'
+    vectorizer = None
+    if args.vectorizer == VectorizerType.TFIDF:
```

Comment:
how about we just pass in the class name from the cmd line and init from string? e.g., https://stackoverflow.com/questions/4821104/dynamic-instantiation-from-string-name-of-a-class-in-dynamically-imported-module

that way we don't need this enum...

Reply:
Interesting. I didn't know this is possible.